### PR TITLE
Makefile tidy up

### PR DESCRIPTION
**Pull request by marcomorain**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       CGO_ENABLED: 0
     steps:
       - checkout
-      - run: gometalinter ./...
+      - run: make lint
```
Comment (zzak) [inline .circleci/config.yml]: That image `supinf/gometalinter` doesn't have make installed :(

Comment (marcomorain) [reply .circleci/config.yml]: boo 🙈

File: Makefile [hunk 2]
```diff
@@ -1,33 +1,14 @@
...
-endif
-
-.PHONY: build/*
-$(BUILD_DIR)/%/amd64/$(EXECUTABLE): $(GOFILES)
-	GOOS=$* GOARCH=amd64 CGO_ENABLED=0 go build $(LDFLAGS) -o $(BUILD_DIR)/$*/amd64/$(EXECUTABLE) .
-
-.PHONY: build-all
-build-all: $(BUILD_DIR)/darwin/amd64/$(EXECUTABLE) $(BUILD_DIR)/linux/amd64/$(EXECUTABLE)
+build/%/amd64/circleci: always
+	GOOS=$* GOARCH=amd64 go build -v -o $@ .
```
Comment (zzak) [inline Makefile]: Am I understanding this correctly, will build both linux and mac binaries every time?

For some convenience, we wanted `make` to just build the current OS's binary to save some time.

**Comment by marcomorain** [comment]
Updated @zzak to add single build back in.

**Comment by zzak** [comment]
@marcomorain `make build-all` produces some strange output for me:

```
make build-all
GOOS=linux GOARCH=amd64 go build -v -o build/linux/amd64/circleci .
GOOS=darwin GOARCH=amd64 go build -v -o build/darwin/amd64/circleci .
errors
runtime/internal/sys
internal/race
runtime/internal/atomic
sync/atomic
internal/cpu
unicode/utf8
runtime
internal/testlog
math
unicode
encoding
unicode/utf16
crypto/subtle
crypto/internal/cipherhw
math/bits
internal/nettrace
container/list
vendor/golang_org/x/crypto/cryptobyte/asn1
vendor/golang_org/x/crypto/poly1305
vendor/golang_org/x/crypto/curve25519
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/hcl/strconv
strconv
crypto/rc4
sync
io
reflect
syscall
math/rand
internal/singleflight
bytes
strings
bufio
path
crypto/cipher
hash
hash/crc32
time
crypto/aes
crypto
crypto/sha512
crypto/hmac
encoding/binary
sort
crypto/md5
crypto/sha1
encoding/base64
internal/poll
crypto/des
crypto/sha256
encoding/pem
vendor/golang_org/x/crypto/chacha20poly1305/internal/chacha20
vendor/golang_org/x/text/transform
os
github.com/CircleCI-Public/circleci-cli/vendor/golang.org/x/sys/unix
vendor/golang_org/x/crypto/chacha20poly1305
regexp/syntax
text/tabwriter
fmt
path/filepath
vendor/golang_org/x/net/route
io/ioutil
regexp
github.com/CircleCI-Public/circleci-cli/vendor/github.com/spf13/afero/mem
github.com/CircleCI-Public/circleci-cli/vendor/golang.org/x/text/transform
html
context
encoding/json
log
github.com/CircleCI-Public/circleci-cli/vendor/github.com/pkg/errors
math/big
mime
mime/quotedprintable
net
compress/flate
github.com/CircleCI-Public/circleci-cli/logger
encoding/hex
net/url
compress/gzip
os/exec
os/user
crypto/rand
crypto/elliptic
encoding/asn1
crypto/rsa
crypto/dsa
crypto/x509/pkix
crypto/ecdsa
vendor/golang_org/x/crypto/cryptobyte
vendor/golang_org/x/net/http2/hpack
vendor/golang_org/x/text/unicode/bidi
vendor/golang_org/x/text/unicode/norm
net/textproto
crypto/x509
mime/multipart
vendor/golang_org/x/text/secure/bidirule
vendor/golang_org/x/net/proxy
net/http/internal
encoding/csv
github.com/CircleCI-Public/circleci-cli/vendor/github.com/fsnotify/fsnotify
vendor/golang_org/x/net/idna
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/hcl/token
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/hcl/ast
crypto/tls
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/hcl/scanner
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/json/token
vendor/golang_org/x/net/lex/httplex
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/hcl/parser
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/json/scanner
flag
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/hcl/printer
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl/json/parser
github.com/CircleCI-Public/circleci-cli/vendor/github.com/hashicorp/hcl
github.com/CircleCI-Public/circleci-cli/vendor/github.com/mitchellh/mapstructure
github.com/CircleCI-Public/circleci-cli/vendor/github.com/pelletier/go-toml
github.com/CircleCI-Public/circleci-cli/vendor/golang.org/x/text/unicode/norm
text/template/parse
net/http/httptrace
text/template
net/http
github.com/CircleCI-Public/circleci-cli/vendor/github.com/spf13/jwalterweatherman
github.com/CircleCI-Public/circleci-cli/vendor/github.com/spf13/pflag
github.com/CircleCI-Public/circleci-cli/vendor/gopkg.in/yaml.v2
html/template
github.com/CircleCI-Public/circleci-cli/settings
github.com/CircleCI-Public/circleci-cli/vendor/github.com/spf13/cast
os/signal
github.com/CircleCI-Public/circleci-cli/vendor/github.com/chzyer/readline
github.com/CircleCI-Public/circleci-cli/vendor/github.com/lunixbochs/vtclean
github.com/CircleCI-Public/circleci-cli/vendor/github.com/juju/ansiterm/tabwriter
github.com/CircleCI-Public/circleci-cli/filetree
github.com/CircleCI-Public/circleci-cli/vendor/github.com/mattn/go-isatty
github.com/CircleCI-Public/circleci-cli/vendor/github.com/mattn/go-colorable
github.com/CircleCI-Public/circleci-cli/vendor/github.com/juju/ansiterm
github.com/CircleCI-Public/circleci-cli/vendor/github.com/manifoldco/promptui/list
github.com/CircleCI-Public/circleci-cli/vendor/github.com/manifoldco/promptui/screenbuf
github.com/CircleCI-Public/circleci-cli/vendor/github.com/spf13/cobra
github.com/CircleCI-Public/circleci-cli/vendor/github.com/manifoldco/promptui
github.com/CircleCI-Public/circleci-cli/vendor/github.com/machinebox/graphql
github.com/CircleCI-Public/circleci-cli/vendor/github.com/magiconair/properties
github.com/CircleCI-Public/circleci-cli/vendor/github.com/spf13/afero
github.com/CircleCI-Public/circleci-cli/client
github.com/CircleCI-Public/circleci-cli/vendor/github.com/spf13/viper
github.com/CircleCI-Public/circleci-cli/api
github.com/CircleCI-Public/circleci-cli/cmd
github.com/CircleCI-Public/circleci-cli
```

**Comment by marcomorain** [comment]
I'm calling `go build -v` that lists each file as it compiles.

**Comment by ndintenfass** [comment]
Anything holding up this merge now?

**Pull request merged**